### ISSUE TITLE
Game Changing Suggestions 25+

### ISSUE THREAD

**Issue by QualityNodeNet** [drtshock/Factions#1054]
FactionsUUID version (run /f version): v1.6.9.5-U0.2.1-SNAPSHOT-b89

Server software (run /version): CraftBukkit version git-PaperSpigot-"8b18730" (MC: 1.8.8) (Implementing API version 1.8.8-R0.1-SNAPSHOT)

Server (logs/latest.log): Nil

**Details:** Hello, I'm not sure if some of these features have already been added, as UUID already offers quite a large amount of features, but these are some suggestions I've had over the past year that I've been recommended, I know some offer these in custom forks, but I see no issue with them being implemented into UUID's official build, especially if there was a way to enable/disable the features some server owner's do not wish to have, anyway here is the list:

**Features:**

1) FactionsFly where you can set perms to where players can specifically fly e.g. Wilderness, Warzone, Truce, Enemy, Ally etc, plus with the Ability for staff to bypass these limitations, and maybe also add an EXP functionality to it, this will need to be optional as some owners wouldn't want this. Basically if a player is flying for one minute it will remove 60 EXP (1 Per second etc), but with permission node setups so different ranks have different amount of EXP takes.

2) Implement Faction Homes, not /f home but when someone has /sethome at faction land, then he leave/got kicked from faction, his sethome must be auto-deleted

3) Implement Territory Logout, configurable, after amount of minutes to teleport player at spawn, for example if you logout at enemy territory and is set to 60 mins (1 hour) and you join back after 1 hour and 1 second or more, you must be teleported to spawn.

4) Add a Cooldown for /f create and /f disband to prevent spamming link this: http://prntscr.com/ig6vo5

**Commands:**

/f shield (Opens a GUI open, where you can purchase shields that last from a configurable amount of time, for a configurable amount of money, these shields will prevent raiding, disabling TnT/Creeper damage, the shields will also have a configurable cooldown to prevent people constantly purchasing them, similar to Clash of Clans shields to be honest)
/f wild (Teleports you to a random spot within the wilderness, with a customisable cooldown and fee, there will be safe checks though to prevent spawning in water, trees or into blocks etc)
/f upgrade (Allow it to be configurable to allow owners to choose what they want players to be able to upgrade and how Money, Tokens, EXP features it will upgrade can be number of Faction Warps, how fast crops grow, or how fast mobs spawn, exp multiplier, a sell multiplier and maybe mcMMO multiplier)
/f checkpoint (Similar to /f home, faction checkpoint allows you to quickly set a checkpoint in either your own faction land, wilderness or warzone.)
/f stealth (Allows you to toggle stealth mode. When you have stealth mode enabled and you are near an enemy faction player in their claim, their fly will not be disabled.)
/f rules (Shows you the set of rules for our factions server)
/f setrule/addrule (Adds rules to /f rules)
/f removerule (Removes rules to /f rules)
/f set coleader <player> (Makes a faction member a coleader)
/f tntbank add [amount/all] - add tnt from your inventory to your factions tntbank
/f tntbank remove [amount] - remove tnt from you factions tntbank to your inventory
/f tntbank chestmode - toggle chestmode, will take take tnt from the chest and add to your inventory
/f tntbank bal [faction_name] - display avaliable tnt in yours/other factions tntbank
/f tntbank tntfill [amount] [radius] - fill all dispensers in [radius] with the specified [amount] from your factions tntbank

**Anti Glitching Features:**
1) Ender Pearl Glitch - this fixes the enderpearl glitch in faction servers, you cannot enderpearl between blocks!

2) Creeper Eggs Glitch - this feature automatically kills creepers when they are spawned into walls 

3) Anti Boat Glitch - prevents glitchers from placing boats in enemy territory!

4) Horse Riding Glitch - disable horse riding so players can't glitch through walls and get into enemy bases.

5) Anti Anvil Sneak - prevents players from using anvils while sneaking!

6) Enemy Pearl Block - prevents players from throwing enderpearls in water in other faction claims.

**Comment by WizzleBear** [comment]
Okay, great stuff but some of it has already been implemented. Just not sure about /f shield, /f tntbank, /f stealth, and /f claimline. It makes FUUID not HCF, and instead more like an annoying knock-off of CosmicPVP.
/f claimchunk is stupid. We have /f claim (radius) (faction).

If you really want some of these features, an add-on has been made [https://www.spigotmc.org/resources/savagefactions-f-fly-f-upgrades-f-checkpoint-f-warp-gui.52891/](url)
For the anti-glitch features, try this plugin: [https://www.spigotmc.org/resources/betterraids-1-8-1-12-faction-server-fix-massivecraft-factions-factionsuuid-silkspawners.48109/](url)

**Comment by QualityNodeNet** [comment]
Thank you, for your reply @WizzleBear however with what you’re saying about it not being HCF, is the reason why I opened with features being optional through enable/disable lines within the config, that resolves all the issues which you raised. Also those four commands you listed as knock offs of ComsicPvP I honestly don’t know if they’re as I haven’t actively played that server in over three years however they’ve been recommended to me by my players and have been added to numerous UUID forks, so obviously they’re wanted features. Also thank you for the links, but I work with a Developer so we don’t need those plugins. This whole post was about trying to remove the need for 5-6 Faction plugins which some servers require as each one offers one more required feature that the previous plugin didn’t offer. Plus to allow Server Owners the chance to offer more features to their players, without needing to purchase forks. I wouldn’t even mind if the price of UUID went up to around 50USD I’d purchase it again, if it had all those features, so would so many other server owners. However whatever features which don’t get added we will fork into our custom version of UUID once 2.0 is stable.

**Comment by ImNachoMans** [comment]
Everything should be configureable since half of these stuff i wouldn't want. Like the /f shield is one of them. I don't want my server to become clash of clans, but I like the idea for all of these.

**Comment by QualityNodeNet** [comment]
Thanks for the reply @NachoMans and yeah I can easily understand how some servers wouldn’t want certain features however others would and to boost UUID’s popularity and make it the best Factions plugin again, adding all these features and allowing Server Owners to choose which features they want to showcase will be very nice step in that direction.

**Comment by QualityNodeNet** [comment]
Added a few more features and italics the features I have noticed are already been implemented.

**Comment by prosavage** [comment]
They called my plugin a cosmic knock off rip.

**Comment by Baerable** [comment]
Most of these features are already implemented, however some are great ideas!

**Comment by QualityNodeNet** [comment]
@ProSavage Yeah apparently a lot of these are in CosmicPvP according to others I've asked however that doesn't mind me, as they're heavy in demand by Server Owners and by my server players themselves, why restrict a players experience just because it's similar to another server, all this does it make it an even playing field, forcing CosmicPvP to push the barrier again and expand with new custom features. @Baerable Yeah I've noticed that and some he's recently added, I've put in italics everything that I've tested and works on my Server (Economy is disabled, so unable to test those features)

**Comment by prosavage** [comment]
Im supposed to start putting all my work into factionsUUID soon

**Comment by QualityNodeNet** [comment]
That's nice @ProSavage You have a great plugin, however I dislike the idea of Faction Servers requiring to run 4-5 different Faction plugins which share numerous similarities but one plugin offers a better feature or a feature which isn't within the other plugin. For the communities sake, I'm just trying to make it easier on them, plus it also allows Trent to potentially gain more sales to help with development. (Plus you plugin over-rides, it doesn't run with it)

Edit: Just fully removed all features which are already implemented cross referenced in-game with the Wiki.

**Comment by Baerable** [comment]
/f set recruit (Makes a faction member a recruit) - implemented, different command
/f set owner (Makes a faction member a owner) - implemented, different command
/f access [f|p|view] [playername/factionname] [all| clear] (Access a player or faction to a certain chunk) - implemented, different command
/f set coleader (Makes a faction member a coleader) - co leaders aren't a part of FactionsUUID yet, but I would love it :)

**Comment by QualityNodeNet** [comment]
Thanks @Baerable I've removed those from the list, plus other features which he has added recently, FactionsFly he's added but not to the extent of my suggestion.
